core/src/components/modal/gestures/sheet.ts

```diff
@@ -233,6 +234,15 @@ export const createSheetGesture = (
      */
     canDismissBlocksGesture = baseEl.canDismiss !== undefined && baseEl.canDismiss !== true && minBreakpoint === 0;
 
+    /**
+     * Cache the scroll element reference when the gesture starts,
+     * this allows us to avoid querying the DOM for the target in onMove,
+     * which would impact performance significantly.
+     */
+    const targetEl = findClosestIonContent(detail.event.target! as HTMLElement);
```

Comment:
Maybe put this logic in an `if (!expandToScroll)`? To skip during gesture start for most scenarios that don't use it.

Reply:
Excellent catch, done!